### ISSUE TITLE
Sync and Fresh Sync can be executed successfully, but ElasticPress still says "There is a problem with connecting to your Elasticsearch host."

### ISSUE THREAD

**Issue by astanabe** [10up/ElasticPress#2685]
I'm now trying to construct WordPress site with Elasticseach in a single machine.
I have installed WordPress 5.9.2, Elasticsearch 7.10.2 and ElasticPress 4.0.1.
The settings of "elasticsearch.yml" is the following.

> cluster.name: foobar
> node.name: foobarnode1
> path.data: /var/lib/elasticsearch
> path.logs: /var/log/elasticsearch
> network.host: 0.0.0.0
> http.host: 0.0.0.0
> http.port: 9200
> discovery.type: single-node
> xpack.security.enabled: false
> http.cors.enabled : true
> http.cors.allow-origin : "*"
> http.cors.allow-methods : OPTIONS, HEAD, GET, POST, PUT, DELETE
> http.cors.allow-headers : X-Requested-With,X-Auth-Token,Content-Type, Content-Length

The output of `curl http://localhost:9200/` is the following.

> {
>   "name" : "foobarnode1",
>   "cluster_name" : "foobar",
>   "cluster_uuid" : "snip",
>   "version" : {
>     "number" : "7.10.2",
>     "build_flavor" : "default",
>     "build_type" : "deb",
>     "build_hash" : "snip",
>     "build_date" : "2021-01-13T00:42:12.435326Z",
>     "build_snapshot" : false,
>     "lucene_version" : "8.7.0",
>     "minimum_wire_compatibility_version" : "6.8.0",
>     "minimum_index_compatibility_version" : "6.0.0-beta1"
>   },
>   "tagline" : "You Know, for Search"
> }

After, I started Elasticsearch service and set `http://localhost:9200/` as "Elasticsearch Host URL" in Settings page of ElassticPress, Sync and Fresh Sync can be executed without error.
However, ElasticPress still says `There is a problem with connecting to your Elasticsearch host.`, and I cannot enable Features of ElasticPress.

I tried .deb version of Elasticsearch 7.10.2, .tar version of Elasticsearch 7.10.2, and .tar version of OpenSearch 1.3.0, but the error cannot be fixed.
How can I fix this error?

**Comment by astanabe** [comment]
Here is a Query Log.

> Type: put_mapping
> Host: http://localhost:9200/
> Time Taken: 14 ms
> URL: http://localhost:9200/foobar-post-1
> Method: PUT
> Headers:
> 
> array(1) {
>   ["Content-Type"]=>
>   string(16) "application/json"
> }
> bool(true)
> 
> Query Body:
> 
> snip
> 
> Query Response Code: HTTP 400
> Query Result:
> 
> null
> 
> Copy cURL Request

**Comment by thomasdiesenreiter** [comment]
Hi @felipeelia, we have the same weird problem with the latest elasticpress version, altthough with wp 5.9.3 (which shouldn't make much of a difference).
indexing works, but I get "Error: Could not connect to Elasticsearch." errors in wordpress and in the cli.
I tried different versions of elasticsearch as well, still not working.
We upgraded from php8.0 to php8.1, maybe this could be part of the problem?
